include/picongpu/particles/atomicPhysics/kernel/CheckForOverSubscription.kernel

```diff
@@ -29,78 +30,73 @@
+            typename T_RejectionProbabilityCacheDataBox>
         HDINLINE void operator()(
             T_Worker const& worker,
             T_AreaMapping const areaMapping,
             T_LocalTimeRemainingBox const timeRemainingBox,
-            T_LocalHistogramDataBox histogramBox,
-            T_LocalElectronHistogramOverSubscribedDataBox localElectronHistogramOverSubscribedBox,
-            T_LocalRejectionProbabilityCacheDataBox localRejectionProbabilityCacheBox) const
+            T_LocalHistogramDataBox const histogramBox,
+            T_EFieldBox const eFieldBox,
+            T_FieldEnergyUseCacheBox const fieldEnergyUseCacheBox,
+            T_SharedRessourcesOverSubscribedBox sharedResourcesOverSubscribedBox,
+            T_RejectionProbabilityCacheDataBox rejectionProbabilityCacheBox) const
         {
-            pmacc::DataSpace<simDim> const superCellIdx = areaMapping.getSuperCellIndex(worker.blockDomIdxND());
-
-            // atomicPhysics superCellFields have no guard, but areMapping includes a guard
-            //  -> must subtract guard to get correct superCellFieldIdx
-            pmacc::DataSpace<picongpu::simDim> const superCellFieldIdx
-                = superCellIdx - areaMapping.getGuardingSuperCells();
+            auto const superCellIdx = KernelIndexation::getSuperCellIndex(worker, areaMapping);
+            auto const superCellFieldIdx = KernelIndexation::getSuperCellFieldIndex(worker, areaMapping, superCellIdx);
 
             auto const timeRemaining = timeRemainingBox(superCellFieldIdx);
-            // end kernel if superCell already finished
             if(timeRemaining <= 0._X)
                 return;
 
-            auto forEachBin = pmacc::lockstep::makeForEach<T_Histogram::numberBins, T_Worker>(worker);
-            auto onlyMaster = pmacc::lockstep::makeMaster<T_Worker>(worker);
-
-            T_Histogram& histogram = histogramBox(superCellFieldIdx);
+            particles::atomicPhysics::localHelperFields::RejectionProbabilityCache<
+                T_Histogram::numberBins,
+                T_FieldEnergyUseCache::numberCells>& rejectionProbabilityCache
+                = rejectionProbabilityCacheBox(superCellFieldIdx);
 
-            uint32_t& histogramOverSubscribed = localElectronHistogramOverSubscribedBox(superCellFieldIdx);
+            bool sharedResourcesOverSubscribed = false;
 
-            picongpu::particles::atomicPhysics::localHelperFields::RejectionProbabilityCache<T_Histogram::numberBins>&
-                rejectionProbabilityCache
-                = localRejectionProbabilityCacheBox(superCellFieldIdx);
-
-            // reset to initial value
-            onlyMaster([&histogramOverSubscribed]() { histogramOverSubscribed = u32(false); });
-            worker.sync();
-
-            /* check all bins for overSubscription
-             *   if yes: calculate RejectionProbability mark entire histogram as over subscribed
-             */
```

Comment:
Your removed the comment here, too?

Reply:
because it was just repeating the source code and the behavior was already documented in the documentation of the stage.